errors.go

```diff
@@ -402,3 +402,44 @@ func IsOAuth2(e error) bool {
 func IsEOF(e error) bool {
 	return Unwrap(e) == io.EOF
 }
+
+// Retry return new instance of RetryError which indicates a potentially recoverable error type
+func Retry(err error, message string, args ...interface{}) error {
+	return WrapWithMessage(&RetryError{
+		Message: fmt.Sprintf(message, args...),
+		Err:     err,
+	}, message, args...)
+}
+
+// RetryError indicates a network related problem
+type RetryError struct {
+	Message string `json:"message"`
+	Err     error  `json:"-"`
+}
+
+// Error is debug - friendly error message
+func (c *RetryError) Error() string {
+	if c.Err == nil {
+		return c.Message
+	}
+	return c.Err.Error()
+}
+
+// IsRetryError indicates that this error is of RetryError type
+func (c *RetryError) IsRetryError() bool {
+	return true
+}
+
+// OrigError returns original error (in this case this is the error itself)
+func (c *RetryError) OrigError() error {
+	return c
+}
+
+// IsRetryError returns whether this error is of ConnectionProblemError
```

Comment:
s/ConnectionProblemError/RetryError
`IsRetryError returns whether this error is of type RetryError`